### PR TITLE
Enable Prometheus to scrape brokered Origins

### PR DESCRIPTION
**Pull request by patrickbrophy**
This PR addresses issue #2823. This PR does the following
- Enable Director to scrape brokered Origins
- Actually start broker if the broker is enabled in the director
- Used Howard's PR to pass a server's prefix to the broker
- Support director only broker usage
- Use HTTPS in broker test to match production behavior

### CODE REVIEW

File: origin/broker_client.go [hunk 1]
```diff
@@ -91,6 +91,14 @@ func getTransport() *http.Transport {
 }
 
 func proxyOrigin(resp http.ResponseWriter, req *http.Request, engine *gin.Engine) {
+	// Handle /metrics endpoint - route to gin engine for Prometheus metrics
+	if req.URL.Path == "/metrics" || strings.HasPrefix(req.URL.Path, "/metrics/") {
```
Comment (h2zh) [inline origin/broker_client.go]: Could you give me the full url examples for the second condition?

File: broker/dialer.go [hunk 2]
```diff
@@ -78,10 +78,11 @@ func NewBrokerDialer(ctx context.Context, egrp *errgroup.Group) *BrokerDialer {
 
 // Set the dialer to use `brokerUrl` as the broker endpoint for
 // the service `name`.
-func (d *BrokerDialer) UseBroker(serverType server_structs.ServerType, name, brokerUrl string) {
+func (d *BrokerDialer) UseBroker(serverType server_structs.ServerType, name, brokerUrl, prefix string) {
 	d.brokerEndpoints.Set(name, brokerPrefixInfo{
 		ServerType: serverType,
 		BrokerUrl:  brokerUrl,
+		Prefix:     prefix,
```
Comment (h2zh) [inline broker/dialer.go]: Actually, after #2785 merged, we can switch `Prefix` to `ServerID` as the component in the `brokerPrefixInfo`.

brokerPrefixInfo.ServerID <-- ServerAd.ServerID <-- OriginAdvertiseV2.ServerID

File: broker/dialer.go [hunk 3]
```diff
@@ -78,10 +78,11 @@ func NewBrokerDialer(ctx context.Context, egrp *errgroup.Group) *BrokerDialer {
 
 // Set the dialer to use `brokerUrl` as the broker endpoint for
 // the service `name`.
-func (d *BrokerDialer) UseBroker(serverType server_structs.ServerType, name, brokerUrl string) {
+func (d *BrokerDialer) UseBroker(serverType server_structs.ServerType, name, brokerUrl, prefix string) {
```
Comment (h2zh) [inline broker/dialer.go]: Tbh, I really like this approach - passing in prefix from server ad instead of recklessly assembling it in this func (given so many edge cases of the `prefix` in prod, it doesn't have a 100% correct pattern)

**Review by h2zh — APPROVED** [review]
Nice PR - concise code, reasonable commits, clear test cases renovation.
